be/src/storage/lake/rowset_update_state.cpp

```diff
@@ -339,22 +460,37 @@ Status RowsetUpdateState::rewrite_segment(const TxnLogPB_OpWrite& op_write, cons
     std::set<uint32_t> update_columns_set(update_colum_ids.begin(), update_colum_ids.end());
     // columns needs to be read from tablet's data
     std::vector<uint32_t> read_column_ids;
-    for (uint32_t i = 0; i < tablet_schema->num_columns(); i++) {
-        if (update_columns_set.find(i) == update_columns_set.end()) {
-            read_column_ids.push_back(i);
+    // may be the implict partial update for auto increment only
+    if (!txn_meta.partial_update_column_ids().empty()) {
+        for (uint32_t i = 0; i < tablet_schema->num_columns(); i++) {
+            if (update_columns_set.find(i) == update_columns_set.end()) {
+                read_column_ids.push_back(i);
+            }
         }
     }
 
+    std::vector<bool> need_rename(rowset_meta->segments_size(), true);
     for (int i = 0; i < rowset_meta->segments_size(); i++) {
         auto src_path = rowset_meta->segments(i);
         auto dest_path = op_write.rewrite_segments(i);
 
         int64_t t_rewrite_start = MonotonicMillis();
-        FooterPointerPB partial_rowset_footer = txn_meta.partial_rowset_footers(i);
-        // if rewrite fail, let segment gc to clean dest segment file
-        RETURN_IF_ERROR(SegmentRewriter::rewrite(tablet->segment_location(src_path),
-                                                 tablet->segment_location(dest_path), *tablet_schema, read_column_ids,
-                                                 _partial_update_states[i].write_columns, i, partial_rowset_footer));
+        if (op_write.txn_meta().has_auto_increment_partial_update_column_id() &&
+            !_auto_increment_partial_update_states[i].skip_rewrite) {
+            RETURN_IF_ERROR(SegmentRewriter::rewrite(
+                    tablet->segment_location(src_path), tablet->segment_location(dest_path), *tablet_schema,
+                    _auto_increment_partial_update_states[i], read_column_ids,
+                    _partial_update_states.size() != 0 ? &_partial_update_states[i].write_columns : nullptr, op_write,
+                    tablet));
+        } else if (_partial_update_states.size() != 0) {
+            FooterPointerPB partial_rowset_footer = txn_meta.partial_rowset_footers(i);
+            // if rewrite fail, let segment gc to clean dest segment file
+            RETURN_IF_ERROR(SegmentRewriter::rewrite(
+                    tablet->segment_location(src_path), tablet->segment_location(dest_path), *tablet_schema,
+                    read_column_ids, _partial_update_states[i].write_columns, i, partial_rowset_footer));
+        } else {
+            need_rename[i] = false;
```

Comment:
If the partial update column is AUTO_INCREMENT column only, and if all rows is NOT exist in the previous version. It means that all AUTO_INCREMENT id allocated in DeltaWriter need not be replace in apply phase. So we do not rewrite the corresponding segment file at all. In this case, we do not rename the file because no rewrite operation for this segment.